sdk/client-core/core/src/main/java/io/clientcore/core/http/client/DefaultHttpClient.java

```diff
@@ -418,18 +419,57 @@ private static void buildAndSend(HttpRequest httpRequest, OutputStreamWriter out
 
         /**
          * Reads the response from the input stream and extracts the information needed to construct an instance of
-         * HttpUrlConnectionResponse
+         * Response
          *
          * @param httpRequest The HTTP Request being sent
          * @param reader the input stream from the socket
-         * @return an instance of HttpUrlConnectionResponse
+         * @return an instance of Response
          * @throws IOException If an I/O error occurs
          */
         private static Response<?> buildResponse(HttpRequest httpRequest, BufferedReader reader) throws IOException {
+            // Parse Http response from socket:
+            // Status Line
+            // Response Headers
+            // Blank Line
+            // Response Body
+
+            int statusCode = readStatusCode(reader);
+
+            HttpHeaders headers = readResponseHeaders(reader);
+            HttpHeader contentLengthHeader = headers.get(CONTENT_LENGTH);
+            String line;
+            // read body if present
+            if (headers.getSize() != 0 && contentLengthHeader != null && contentLengthHeader.getValue() != null) {
+                StringBuilder bodyString = new StringBuilder();
+                while ((line = reader.readLine()) != null) {
+                    bodyString.append(line);
+                }
+
+                return new HttpResponse<>(httpRequest, statusCode, headers,
+                    BinaryData.fromString(bodyString.toString()));
+            }
+            return new HttpResponse<>(httpRequest, statusCode, headers, null);
+        }
+
+        private static int readStatusCode(BufferedReader reader) throws IOException {
+
+            while (!reader.ready()) {
```

Comment:
There's no termination here if the reader never becomes ready. We should add a response timeout handler here to make sure we don't infinite loop in a scenario where a server never responds.

Reply:
Do we default responseTimeout that we use and does this mean we just block this on a thread and wait for this task to complete within the specified timeout?